### DISCUSSION TITLE
How to access gocron.Job

### DISCUSSION THREAD

**Discussion by justpoypoy** [go-co-op/gocron#796] [Q&A]
How to access job interfaces eg. job.LastRun(), job.ID() inside task

this my example code
```
	repo := repositories.New(r.di)
	services := usecases.New(repo)
	handle := handler.New(services)
_, err := s.NewJob(
		gocron.DailyJob(
			1,
			gocron.NewAtTimes(
				gocron.NewAtTime(17, 27, 30),
				gocron.NewAtTime(17, 27, 40),
				gocron.NewAtTime(17, 27, 50),
			),
		),
		gocron.NewTask(handle.GetDailyTransaction),
		gocron.WithSingletonMode(gocron.LimitModeWait),
	)
```
	
and how to access unique id or last run inside function GetDailyTransaction

**Comment by JohnRoesler** [comment]
hi @justpoypoy 

Have you tried setting the job object no the call to NewJob and then referencing it?

```go
	repo := repositories.New(r.di)
	services := usecases.New(repo)
	handle := handler.New(services)
	job, err := s.NewJob(
		gocron.DailyJob(
			1,
			gocron.NewAtTimes(
				gocron.NewAtTime(17, 27, 30),
				gocron.NewAtTime(17, 27, 40),
				gocron.NewAtTime(17, 27, 50),
			),
		),
		gocron.NewTask(handle.GetDailyTransaction(job.ID())),
		gocron.WithSingletonMode(gocron.LimitModeWait),
	)
```

**Reply by justpoypoy** [reply]
hi @JohnRoesler sure, have been tried but job object is return from NewJob right? so i get error undefined job

<img width="724" alt="image" src="https://github.com/user-attachments/assets/beb75e57-5ea5-458a-b314-a1520878d4b6">

**Reply by JohnRoesler** [reply]
ah, yes right. Well, for the UUID, you can set your own, and then pass that variable in 

https://pkg.go.dev/github.com/go-co-op/gocron/v2#WithIdentifier

**Reply by justpoypoy** [reply]
yes if i want parse the UUID, but i want get the instance of gocron.Job like the v1 i used before.

```
s := gocron.NewScheduler(time.Local)
s.Every(1).At("03:00").Day().Saturday().Sunday().Do(handle.GetTransactionsHandler)

s.StartBlocking()
```

and consumed by handler like this

```
func (h *handler) GetTransactionsHandler(job gocron.Job) error {
        fmt.Println(job.LastRun())
	return nil
}
```

**Comment by JohnRoesler** [comment]
@justpoypoy you CAN do something like this, but it's not really that helpful as last run is going to be the same as time.Now

```go
package main

import (
	"log/slog"
	"time"

	"github.com/go-co-op/gocron/v2"
)

func main() {
	scheduler, err := gocron.NewScheduler()
	if err != nil {
		panic(err)
	}

	var job gocron.Job

	job, err = scheduler.NewJob(
		gocron.DurationJob(time.Second),
		gocron.NewTask(func() {
			lastRun, err := job.LastRun()
			if err != nil {
				slog.Error("job run", slog.String("error", err.Error()))
			} else {
				slog.Info("job run", slog.String("last_run", lastRun.String()))
			}
		}),
	)
	if err != nil {
		panic(err)
	}

	scheduler.Start()
	for {
	}
}
```

If I understand your request, you want to be able to take some action based on the last time the job ran inside of your task function? Could you help me understand the use case a little better?

**Reply by justpoypoy** [reply]
thanks for helping, my usecase is getting time of last running scheduler and use it for query to database
